File: packages/maplibre/android/src/main/java/io/capawesome/capacitorjs/plugins/maplibre/classes/events/MarkerClickEvent.java
package io.capawesome.capacitorjs.plugins.maplibre.classes.events;

import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.maplibre.MapLibreHelper;
import io.capawesome.capacitorjs.plugins.maplibre.interfaces.Result;
import org.maplibre.android.geometry.LatLng;

public class MarkerClickEvent implements Result {

    @NonNull
    private final LatLng coordinates;

    @NonNull
    private final String mapId;

    @NonNull
    private final String markerId;

    private final double x;
    private final double y;

    public MarkerClickEvent(@NonNull LatLng coordinates, @NonNull String mapId, @NonNull String markerId, double x, double y) {
        this.coordinates = coordinates;
        this.mapId = mapId;
        this.markerId = markerId;
        this.x = x;
        this.y = y;
    }

    @Override
    @NonNull
    public JSObject toJSObject() {
        JSObject point = new JSObject();
        point.put("x", x);
        point.put("y", y);
        JSObject result = new JSObject();
        result.put("coordinates", MapLibreHelper.createLatLngObject(coordinates));
        result.put("mapId", mapId);
        result.put("markerId", markerId);
        result.put("point", point);
        return result;
    }
}

File: packages/maplibre/ios/Plugin/Classes/Events/MarkerClickEvent.swift
import Capacitor
import CoreGraphics
import Foundation

@objc public class MarkerClickEvent: NSObject, Result {
    let coordinates: LatLng
    let mapId: String
    let markerId: String
    let point: CGPoint

    init(coordinates: LatLng, mapId: String, markerId: String, point: CGPoint) {
        self.coordinates = coordinates
        self.mapId = mapId
        self.markerId = markerId
        self.point = point
    }

    @objc public func toJSObject() -> AnyObject {
        var pointResult = JSObject()
        pointResult["x"] = Double(point.x)
        pointResult["y"] = Double(point.y)
        var result = JSObject()
        result["coordinates"] = coordinates.toJSObject()
        result["mapId"] = mapId
        result["markerId"] = markerId
        result["point"] = pointResult
        return result as AnyObject
    }
}
